war/src/test/java/io/lumeer/engine/rest/permissions/ProjectServicePermissionsIntegrationTest.java

```diff
@@ -0,0 +1,319 @@
+      assertThat(securityFacade.hasProjectRole(projectCode, LumeerConst.Security.ROLE_READ)).isTrue();
+      assertThat(securityFacade.hasProjectRole(projectCode, LumeerConst.Security.ROLE_WRITE)).isFalse();
+      assertThat(securityFacade.hasProjectRole(projectCode, LumeerConst.Security.ROLE_MANAGE)).isFalse();
+
+      Response response = client.target(TARGET_URI).path(PATH_PREFIX + projectCode).
+            request(MediaType.APPLICATION_JSON).buildGet().invoke();
+      assertThat(response.getStatusInfo()).isEqualTo(Response.Status.OK);
+      client.close();
+      createClient();
+      response = client.target(TARGET_URI).path(PATH_PREFIX + projectCode + "/name").
+            request(MediaType.APPLICATION_JSON).buildGet().invoke();
+      assertThat(response.getStatusInfo()).isEqualTo(Response.Status.OK);
+   }
+
+   @Test
+   public void testUpdateProjectNoRole() throws UserAlreadyExistsException {
+      String groupName = "TestGroup3";
+      userGroupFacade.addGroups(organizationCode, groupName);
+      userGroupFacade.addUser(organizationCode, userEmail, groupName);
+
+      String projectCode = "ProjectServiceTestProject_id";
+      String projectName = "ProjectServiceTestProject";
+      projectFacade.createProject(new Project(projectCode, projectName));
+
+      securityFacade.removeProjectUsersRole(projectCode, Collections.singletonList(userEmail), LumeerConst.Security.ROLE_READ);
+      securityFacade.removeProjectUsersRole(projectCode, Collections.singletonList(userEmail), LumeerConst.Security.ROLE_WRITE);
+      securityFacade.removeProjectUsersRole(projectCode, Collections.singletonList(userEmail), LumeerConst.Security.ROLE_MANAGE);
+      assertThat(securityFacade.hasProjectRole(projectCode, LumeerConst.Security.ROLE_READ)).isFalse();
+      assertThat(securityFacade.hasProjectRole(projectCode, LumeerConst.Security.ROLE_WRITE)).isFalse();
+      assertThat(securityFacade.hasProjectRole(projectCode, LumeerConst.Security.ROLE_MANAGE)).isFalse();
+
+      String newProjectName = "NewName";
+      Response response = client.target(TARGET_URI).path(PATH_PREFIX + projectCode + "/name/" + newProjectName).
+            request(MediaType.APPLICATION_JSON).buildPut(Entity.entity(null, MediaType.APPLICATION_JSON)).invoke();
+      assertThat(response.getStatusInfo()).isEqualTo(Response.Status.UNAUTHORIZED);
+      client.close();
+      createClient();
+      String newprojectCode = "NewCode";
+      String newProjectName2 = "NewName2";
+      response = client.target(TARGET_URI).path(PATH_PREFIX + projectCode).
+            request(MediaType.APPLICATION_JSON).buildPut(Entity.json(new Project(newprojectCode, newProjectName2))).invoke();
+      assertThat(response.getStatusInfo()).isEqualTo(Response.Status.UNAUTHORIZED);
+   }
+
+   @Test
+   public void testGetUpdateProjectManageRole() throws UserAlreadyExistsException {
```

Comment:
This is probably a type and `Get` should be removed from the method name, or not?